vimeo-networking/src/main/java/com/vimeo/networking/model/VideoFile.java

```diff
@@ -104,9 +173,54 @@ public boolean isVP6() {
         return getType() == MimeType.VP6;
     }
 
-    /** @return true if this VideoFile doesn't have an expired field or if the expires date is before the current date */
-    public boolean isExpired() {
-        // If expires is null, we should probably refresh the video object regardless [KV] 3/31/16
-        return expires == null || expires.before(new Date());
+    public int getWidth() {
+        return width;
+    }
+
+    public int getHeight() {
+        return height;
+    }
+
+    public long getSize() {
+        return size;
+    }
+
+    public int getFps() {
+        return fps;
+    }
+
+    @Nullable
+    public String getMd5() {
+        return md5;
+    }
+
+    @Nullable
+    public Date getCreatedTime() {
+        return createdTime;
+    }
+    // </editor-fold>
+
+    // -----------------------------------------------------------------------------------------------------
+    // Equals/Hashcode
+    // -----------------------------------------------------------------------------------------------------
+    // <editor-fold desc="Equals/Hashcode">
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || !(o instanceof VideoFile)) {
+            return false;
+        }
+
+        VideoFile that = (VideoFile) o;
+
+        return (this.getLink() != null && that.getLink() != null) && this.getLink().equals(that.getLink());
```

Comment:
I think that the equals/hashCode implementation here and elsewhere in our models are technically incorrect. We should probably button all of them down since this is a public library. The issues discussed below are not just in this class, but all model classes in networking that I have seen.

1) equals id should never be defined based on a mutable field.
2) we might want to add a canEquals method to handle proper inheritance behavior wrt equality

Also, if both objects have a null link, they are not equal? Just want to double check that this is the behavior that we want.

This article goes over some of the fine points:
http://www.artima.com/lejava/articles/equality.html